app.js

```diff
@@ -1 +1,84 @@
+const usersList = 
+[
+    {
+        "id":0,
+        "userName":"Tor",
+        "userAge":6456,
+        "userPhone":"0959227735",
+    },
+    {
+        "id":1,
+        "userName":"Loki",
+        "userAge":6786,
+        "userPhone":"0959223367",
+    },
+];
 
+const UserRender = ({userData})=>
+(
+    <tr>
+        <td>{userData.userName}</td>
+        <td>{userData.userAge}</td>
+        <td>{userData.userPhone}</td>
+    </tr>
+);
+
+const TableRender = ({usersData})=>
+(
+    <div>
+        <table border = "1px black">
+                <thead>
+                    <tr>
+                        <th>UserName</th>
+                        <th>UserAge</th>
+                        <th>UserPhone</th>
+                    </tr>
+                </thead>
+                <tbody>
+                    {
+                        usersData.map(user => <UserRender key={user.id} userData = {user} />)
+                    }
+               </tbody>
+        </table>
+    </div>
+);
+
+const FormForUSer =({onSubmitFunc})=>
+{
+    const[newUser,setNewValues] = React.useState(
+    {
+        userName:"",
+        userAge:0,
+        userPhone:"",
+    });
+    return(    
+        <form onSubmit = {event=>
+                {
+                    event.preventDefault();
+                    onSubmitFunc({...newUser, id:Date.now()});
+                }
+            }
+        >
+            <input placeholder='User Name' onChange={event => newUser.userName = event.target.value }/>
```

Comment:
values in state should be only updated via callback obtained from useState, not directly. like this:
```suggestion
            <input placeholder='User Name' onChange={event => setNewValues({...newUser, userName: event.target.value}) }/>
```

Reply:
Done